### PR TITLE
[Add] Toast

### PR DESCRIPTION
**Pull request by siva2204**
- add custom toast with `success`, `warn`, `error`, `info` styling

### CODE REVIEW

**Comment by siva2204** [comment]
in `dse` toast throws exception

**Comment by dhruv1294** [comment]
Y? is it due to bottom sheet :(

**Comment by siva2204** [comment]
> Y? is it due to bottom sheet :(

ig its something to do with `context`, have to check

**Comment by Sudhindra3000** [comment]
Can you attach screenshots for the different types of snackbars

**Comment by siva2204** [comment]
Screenshot from 2022-02-10 14-57-24
Screenshot from 2022-02-10 14-56-56
Screenshot from 2022-02-10 14-56-22

Screenshot from 2022-02-10 14-55-47

File: lib/utils/snackbar.dart [hunk 1]
```diff
@@ -1,10 +1,55 @@
+import 'package:dalal_street_client/models/snackbar_props.dart';
+import 'package:dalal_street_client/theme/colors.dart';
+import 'package:another_flushbar/flushbar.dart';
 import 'package:flutter/material.dart';
 
+enum SnackBarType { success, error, info, warning }
```
Comment (Sudhindra3000) [inline lib/utils/snackbar.dart:6]: Create a `snackbar` folder in models, and have this and `SnackbarProps` there

File: lib/utils/snackbar.dart [hunk 2]
```diff
@@ -1,10 +1,55 @@
...
-void showSnackBar(
-  BuildContext context,
-  String msg,
-) =>
-    ScaffoldMessenger.of(context)
-      ..hideCurrentSnackBar()
-      ..showSnackBar(SnackBar(content: Text(msg)));
+void showSnackBar(BuildContext context, String msg,
+    {SnackBarType type = SnackBarType.info}) {
+  var snackBarProps = getSnackBarProps(type);
```
Comment (Sudhindra3000) [inline lib/utils/snackbar.dart:11]:
```suggestion
  final snackBarProps = getSnackBarProps(type);
```

File: lib/utils/snackbar.dart [hunk 3]
```diff
@@ -1,10 +1,55 @@
...
+    case SnackBarType.info:
+      snackBarProps = SnackBarProps('Info', blue, Icons.info_outline);
+      break;
+    case SnackBarType.error:
+      snackBarProps = SnackBarProps('Error', red, Icons.error_outline);
+      break;
+    case SnackBarType.warning:
+      snackBarProps = SnackBarProps('Warning', gold, Icons.warning);
+      break;
+  }
```
Comment (Sudhindra3000) [inline lib/utils/snackbar.dart:52]: Can return props in switch case itself?

**Review by Sudhindra3000 — CHANGES_REQUESTED** [review]

**Pull request merged**